### PR TITLE
planner: try to match universal bindings when optimization

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #48875

Problem Summary: planner: try to match universal bindings when optimization

### What changed and how does it work?

1. Introduce a new variable to control whether to enable universal binding;
2. Try to match universal binding when optimization;

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: pkg/bindinfo/BUILD.bazel [hunk 1]
```diff
@@ -58,7 +58,7 @@ go_test(
     embed = [":bindinfo"],
     flaky = True,
     race = "on",
-    shard_count = 41,
+    shard_count = 43,
```
Comment (elsa0520) [inline pkg/bindinfo/BUILD.bazel:61]: Where is 42 ?

**Review by AilinKid — APPROVED** [review]

**Review by elsa0520 — APPROVED** [review]
LGTM

File: pkg/bindinfo/BUILD.bazel (hunk 1 above)
Comment (qw4990) [reply pkg/bindinfo/BUILD.bazel:61]: Because I added 2 new test cases in `pkg/bindinfo/global_handle_test.go`, it becomes from 41 to 43.

**Pull request merged**